cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -2035,7 +2035,7 @@ Result<std::shared_ptr<ChunkedArray>> TakeCC(const ChunkedArray& values,
     ARROW_ASSIGN_OR_RAISE(new_chunks[i],
                           Concatenate(current_chunk->chunks(), ctx->memory_pool()));
   }
-  return std::make_shared<ChunkedArray>(std::move(new_chunks));
+  return std::make_shared<ChunkedArray>(std::move(new_chunks), values.type());
```

Comment:
Thanks for fixing this!

There's a similar case down on line 2051 here - that should probably be fixed too.

We should also add test cases to make sure this doesn't regress.

Reply:
Thanks for the review @lidavidm I'll address your comments cc @westonpace